include/opensm/osm_opensm.h

```diff
@@ -113,7 +113,8 @@ typedef enum _osm_routing_engine_type {
 	OSM_ROUTING_ENGINE_TYPE_NUE,
 	OSM_ROUTING_ENGINE_TYPE_SSSP,
 	OSM_ROUTING_ENGINE_TYPE_DFSSSP,
-	OSM_ROUTING_ENGINE_TYPE_UNKNOWN
+	OSM_ROUTING_ENGINE_TYPE_UNKNOWN,
+	OSM_ROUTING_ENGINE_TYPE_EXTERNAL = 100
```

Comment:
Note: I set the value to the arbitrary number 100 in case more built-in routing engines will be added in the future.

Reply:
fixed in commit 9787fb4b253e2623d78f258de7922e41a6fa4adc